### PR TITLE
[WIP]Adding option to calculate for a second (or buy-to-let) home

### PR DESCRIPTION
**Pull request by urfolomeus**
Currently the Stamp Duty Calculator only calculates for people that only own a single property. Although the copy had been updated to mention the Stamp Duty tax changes that came into force on 1st April this year, there was no way to let the tool know that you were calculating for additional properties.

This PR introduces that feature.

(The feature was added for both JavaScript and non-JavaScript pages, but the gifs for the non-JavaScript version were too large on account of the colossal amount of time it seems to take to page refresh locally 😁)

#### Before

before

#### After

after

### CODE REVIEW

File: spec/javascripts/unit/services/stampDuty.js [hunk 1]
```diff
@@ -74,5 +89,75 @@ describe('Service: StampDuty', function () {
...
     });
   });
+
+  describe('Stamp Duty Bands - second home', function() {
+    beforeEach(function () {
+      setSecondHome(true);
+    });
+
+    it('when house price is 0', function() {
+      setPrice(0);
```
Comment (paulanthonywilson) [inline spec/javascripts/unit/services/stampDuty.js]: I'll buy that house!

Comment (tomas-stefano) [inline spec/javascripts/unit/services/stampDuty.js]: I want it too!

File: spec/models/stamp_duty_spec.rb [hunk 2]
```diff
@@ -5,12 +5,28 @@
...
+  end
+
+  describe 'setting second home' do
+    it 'is false if "false" is given' do
+      expect(described_class.new(second_home: "false").second_home).to be_falsy
+    end
+
+    it 'is true if "true" is given' do
+      expect(described_class.new(second_home: "true").second_home).to be_truthy
+    end
```
Comment (tomas-stefano) [inline spec/models/stamp_duty_spec.rb]: Can we split it up into contexts? And can we add a '#' to the describe?

Comment (urfolomeus) [inline spec/models/stamp_duty_spec.rb]: Will do.

**Comment by urfolomeus** [comment]
@tomas-stefano done

File: spec/models/stamp_duty_spec.rb [hunk 3]
```diff
@@ -24,57 +50,161 @@
...
+      end
     end
 
     context 'when house price is 125000' do
       let(:price) { 125000 }
 
-      its(:tax_due) { is_expected.to be_zero }
-      its(:percentage_tax) { is_expected.to be_zero }
-      its(:total_due) { is_expected.to eql(125000) }
+      context 'and is not a second home' do
```
Comment (Tr4pSt3R) [inline spec/models/stamp_duty_spec.rb]: What if it's a third home? Or an _n-th_ home?

Comment (urfolomeus) [inline spec/models/stamp_duty_spec.rb]: Same rules apply. It's just down to whether or not you have the added checkbox ticked or not.

**Comment by Tr4pSt3R** [comment]
I can see any more holes to poke :)
All the RoR stuff looks good
👍 🚢

**Comment by urfolomeus** [comment]
Thanks @Tr4pSt3R :)

**Comment by paulanthonywilson** [comment]
👍

**Pull request merged**